docs/root/configuration/best_practices/edge.rst

```diff
@@ -11,19 +11,19 @@ TCP proxies should configure:
 
 * restrict access to the admin endpoint,
 * :ref:`overload_manager <config_overload_manager>`,
-* :ref:`listener buffer limits <envoy_api_field_Listener.per_connection_buffer_limit_bytes>` to 32 KiB,
-* :ref:`cluster buffer limits <envoy_api_field_Cluster.per_connection_buffer_limit_bytes>` to 32 KiB.
+* :ref:`listener buffer limits <envoy_v3_api_field_config.listener.v3.Listener.per_connection_buffer_limit_bytes>` to 32 KiB,
```

Comment:
Apart from the conversation of how often we want to do an API upgrade (https://github.com/envoyproxy/envoy/issues/10852), any thoughts on making this less painful in the future? Could we somehow have sphinx warn/error on using a ref that is not the current API version? Should we somehow anchor this to an API that would "move" to the current version and error otherwise? I'm just trying to figure out ways of making this less manual. If you have any thoughts can we make sure we have a docs issue tracking them?

Reply:
A floating "anchor" would solve some issues, but not all (type urls for example). I automated ~70%-80% of the work updating the references, enums, and type urls; more automation is possible too. Updates to deprecated fields in examples are going to be trickier: some can be done via templated replacements. Switches to ```typed_config``` fields would require quite a bit of work to automate. Perhaps we need to break out examples and evolve them together with code changes? This might make them testable too.

Reply:
I do think we need to break out the examples into something testable in the future. Doesn't have to be this PR though.